File: home/ubuntu/remitflow/54remitflow-unified-platform/backend/models/go/journeys/journey_16_wallet_topup_models.go
// Wallet Top-up Database Models
// Journey: journey_16_wallet_topup
// GORM ORM Models

package models

import (
	"time"
	"gorm.io/gorm"
)


type WalletTopup struct {
	gorm.Model
	UserID    uint           `gorm:"not null;index" json:"user_id"`
	Status    string         `gorm:"size:50;default:pending" json:"status"`
	Metadata  string         `gorm:"type:jsonb" json:"metadata"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

func (m *WalletTopup) TableName() string {
	return "wallettopups"
}

File: home/ubuntu/remitflow/54remitflow-unified-platform/backend/models/journeys/journey_16_wallet_topup_models.py
"""
Wallet Top-up Database Models
Journey: journey_16_wallet_topup
SQLAlchemy ORM Models
"""

from sqlalchemy import Column, Integer, String, DateTime, Boolean, Float, ForeignKey, JSON
from sqlalchemy.orm import relationship
from datetime import datetime

from app.database import Base


class WalletTopup(Base):
    __tablename__ = "wallettopups"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    status = Column(String(50), default="pending")
    metadata = Column(JSON)
    
    # Relationships
    user = relationship("User", back_populates="wallettopups")
    
    def to_dict(self):
        return {
            "id": self.id,
            "user_id": self.user_id,
            "created_at": self.created_at.isoformat(),
            "updated_at": self.updated_at.isoformat(),
            "status": self.status,
            "metadata": self.metadata
        }
